### PR TITLE
Migrating "Addons" to generic handler

### PR DESCRIPTION
**Pull request by bfoley13**
# Description

This migrates the ingress template to the generic handler.  There are some CLI specific logic that will be moved to cmdhelpers for now until we plan out a path for better CLI handling.

The necessary fields were added to the ingress draft.yaml with the defaults being added to keep the same CLI experience.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Refactor

## How Has This Been Tested?

Provide instructions so we can reproduce, and list any relevant details for your test configuration. Please mention if this is a breaking change which will impact consuming tool(s).

- [x] Unit Test:
- [x] E2E Test:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/cmdhelpers/addon_helpers.go [hunk 1]
```diff
@@ -1,65 +1,77 @@
-package addons
+package cmdhelpers
```
Comment (bfoley13) [inline pkg/cmdhelpers/addon_helpers.go:1]: the cmdhelpers package will exist for this and the Workflow CLI functionality for now.  We will plan and migrate this logic somewhere more beneficial to templates/cli integration later on

File: cmd/update.go [hunk 2]
```diff
@@ -60,40 +58,45 @@ func newUpdateCmd() *cobra.Command {
...
-			return err
-		}
-		uc.addon = addon
+	if dryRun {
+		dryRunRecorder = dryrunpkg.NewDryRunRecorder()
+		uc.templateVariableRecorder = dryRunRecorder
+		uc.templateWriter = dryRunRecorder
 	}
 
-	addonConfig, err := addons.GetAddonConfig(template.Addons, uc.provider, uc.addon)
```
Comment (bosesuneha) [inline cmd/update.go:71]: Are we not going to use user provided addon name and only support app routing for now?

Comment (bfoley13) [reply cmd/update.go:71]: Yes, thats the only one we have and we can migrate this correctly to something better in the CLI in the future!

**Review by bosesuneha — APPROVED** [review]
lgtm

File: pkg/handlers/template_test.go [hunk 3]
```diff
@@ -321,6 +321,22 @@ func TestTemplateHandlerValidation(t *testing.T) {
 				"CLUSTERNAME":            "testcluster",
 			},
 		},
+		{
```
Comment (davidgamero) [inline pkg/handlers/template_test.go:324]: as this test grows, breaking it into multiple files would make sense to organize the cases by template

**Review by davidgamero — APPROVED** [review]

File: pkg/handlers/template_test.go (hunk 3 above)
Comment (bfoley13) [reply pkg/handlers/template_test.go:324]: Agreed! We can pull out the testing function for resuse

**Pull request merged**